service/resource/cloudformation/error.go

```diff
@@ -15,3 +19,15 @@ var notFoundError = microerror.New("not found")
 func IsNotFound(err error) bool {
 	return microerror.Cause(err) == notFoundError
 }
+
+// IsStackNotFound asserts stack not found error from upstream's API message
+//
+// FIXME: The validation error returned by the CloudFormation API doesn't make
+// things easy to check, other than looking for the returned string. There's no
+// constant in aws go sdk for defining this string, it comes from the service.
+func IsStackNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), "does not exist")
```

Comment:
Lets please use the cause (`microerror.Cause(err)`) as base for error inspection. We might wrap errors and screw around with the messages through the stack.